libraries/Microsoft.Bot.Builder.AI.LuisV3/LuisPredictionOptions.cs

```diff
@@ -0,0 +1,91 @@
+    /// </summary>
+    public class LuisPredictionOptions
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether all intents come back or only the top one.
+        /// </summary>
+        /// <value>
+        /// True for returning all intents.
+        /// </value>
+        public bool IncludeAllIntents { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not instance data should be included in response.
+        /// </summary>
+        /// <value>
+        /// A value indicating whether or not instance data should be included in response.
+        /// </value>
+        public bool IncludeInstanceData { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether API results should be included.
+        /// </summary>
+        /// <value>True to include API results.</value>
+        /// <remarks>This is mainly useful for testing or getting access to LUIS features not yet in the SDK.</remarks>
+        public bool IncludeAPIResults { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether queries should be logged in LUIS.
+        /// </summary>
+        /// <value>
+        /// If queries should be logged in LUIS in order to help build better models through active learning.
+        /// </value>
+        public bool Log { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets dynamic lists used to recognize entities for a particular query.
+        /// </summary>
+        /// <value>
+        /// Dynamic lists of things like contact names to recognize at query time.
+        /// </value>
+        public IList<DynamicList> DynamicLists { get; set; }
+
+        /// <summary>
+        /// Gets or sets external entities recognized in the query.
+        /// </summary>
+        /// <value>
+        /// External entities recognized in query.
+        /// </value>
+        public IList<ExternalEntity> ExternalEntities { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether external entities should override other means of recognizing entities.
+        /// </summary>
+        /// <value>
+        /// Boolean for if external entities should be preferred to the results from LUIS models.
+        /// </value>
+        public bool PreferExternalEntities { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether production (false) or staging (true) slot should be used.
+        /// </summary>
+        /// <value>
+        /// True for staging or false for production.
+        /// </value>
+        /// <remarks> 
+        /// LUIS supports both a production slot and a staging slot and this controls which is used.
+        /// If you specify a Version, then a private versioned slot is used instead.
+        /// </remarks>
+        public bool Staging { get; set; } = false;
```

Comment:
This is against why we decided to make the slot name part of the URL. If we decide to introduce slots other than the "production" and "staging" slots we have right now, the SDK won't be able to handle them.

Reply:
I had made this a string, but I had two issues:
1) The UI currently only allows production/staging and we need a way to communicate this.  I can set it to "production" by default, but developers won't know about "staging".  I could make some static string definitions to overcome this.
2) I had trouble coming up with scenarios where people that needed more wouldn't use version instead--do you have some?  Several people have already been confused on slots vs. versions and why there are both.

Reply:
I went ahead and made this a string with constants.

Reply:
We have several plans in the future for versioning in LUIS so probably we'll support multiple slots in the future. Anyway, using slots vs versions, is that you don't have to change the URL when you decide to publish a new version. You could publish to a specific slot and the code you'll use it on the fly.